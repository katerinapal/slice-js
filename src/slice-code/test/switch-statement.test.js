import {runAllCombosTests} from './helpers/utils'

// test.only(
//   'switchWithFallThrough("purple")',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/switch-statement'), ({switchWithFallThrough}) => {
//     return [switchWithFallThrough('purple')]
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/switch-statement'),
  methods: [
    {
      methodName: 'switchWithFallThrough',
      possibleArguments: [
        ['green'], ['blue'], ['purple'],
        ['yellow'], ['orange'], ['red'],
      ],
    },
    {
      methodName: 'switchWithDefault',
      possibleArguments: [
        ['twix'], ['snickers'],
      ],
    },
  ],
})
