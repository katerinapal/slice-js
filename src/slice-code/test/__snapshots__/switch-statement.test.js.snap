exports[`test switchWithDefault("snickers") 1`] = `
"export { switchWithDefault };

function switchWithDefault() {
  return \'default, no candy :-(\';
}"
`;

exports[`test switchWithDefault("twix") && switchWithDefault("snickers") 1`] = `
"export { switchWithDefault };

function switchWithDefault(candy) {
  switch (candy) {
    case \'twix\':
      return \`case \${candy}\`;
    default:
      return \'default, no candy :-(\';
  }
}"
`;

exports[`test switchWithDefault("twix") 1`] = `
"export { switchWithDefault };

function switchWithDefault(candy) {
  return \`case \${candy}\`;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("purple") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("blue") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("purple") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("blue") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("purple") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("green") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'green\':
    case \'blue\':
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;

    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") && switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'purple\':
      ret = \`1st \${color}\`;
      break;
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("purple") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;

  ret = \`1st \${color}\`;

  return ret;
}"
`;

exports[`test switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;

  ret = \`2nd \${color}\`;

  return ret;
}"
`;

exports[`test switchWithFallThrough("yellow") && switchWithFallThrough("orange") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("yellow") && switchWithFallThrough("orange") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("yellow") && switchWithFallThrough("red") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithFallThrough("yellow") 1`] = `
"export { switchWithFallThrough };

function switchWithFallThrough(color) {
  let ret = null;
  switch (color) {
    case \'yellow\':
    case \'orange\':
    case \'red\':
      ret = \`2nd \${color}\`;
  }
  return ret;
}"
`;

exports[`test switchWithSideEffects({"name":"harry"}) && switchWithSideEffects({"name":"ron"}) 1`] = `
"export { switchWithSideEffects };

function switchWithSideEffects(character) {
  switch (sideEffect(character)) {
    case \'harry\':
      return [\'main\', character];
    default:
      return [\'supporting\', character];
  }

  function sideEffect(obj) {
    obj.touched = true;
    return obj.name;
  }
}"
`;

exports[`test switchWithSideEffects({"name":"harry"}) 1`] = `
"export { switchWithSideEffects };

function switchWithSideEffects(character) {
  (function (obj) {
    obj.touched = true;
    return obj.name;
  })(character);
  return [\'main\', character];
}"
`;

exports[`test switchWithSideEffects({"name":"ron"}) 1`] = `
"export { switchWithSideEffects };

function switchWithSideEffects(character) {
  (function (obj) {
    obj.touched = true;
    return obj.name;
  })(character);
  return [\'supporting\', character];
}"
`;
